src/odemis/driver/npmc.py

```diff
@@ -0,0 +1,816 @@
+# -*- coding: utf-8 -*-
+'''
+Created on 19 June 2018
+
+@author: Anders Muskens
+
+Copyright © 2018 Anders Muskens, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms
+of the GNU General Public License version 2 as published by the Free Software
+Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+'''
+from __future__ import division
+from concurrent.futures import CancelledError, TimeoutError
+import fcntl
+import glob
+import serial
+import logging
+import os
+import threading
+import time
+import re
+from Pyro4.core import isasync
+from odemis import model
+from odemis.util import driver
+from odemis.model import HwError, CancellableFuture, ParallelThreadPoolExecutor
+
+# Unit definitions 
+UNIT_DEF = {
+    'mm': 2,
+    'um': 3,
+    'in': 4,
+    }
+
+C = 2.99792458e11  # speed of light in mm/s
```

Comment:
Since the device operates in millimetres (and cannot be set to metres), I thought it was better to keep all device constants in mm/s rather than converting all the time and possibly causing confusion. Is it ok to keep this as mm?

Reply:
See later my comments on the conv_factor... I think it'd be even better to not even have the speed of light hard-coded in the driver. It should just be "a factor" coming from the argument/YAML file.